### ISSUE TITLE
Keywords in SymbolTokens are currently not writable

### ISSUE THREAD

**Issue by sharanrokx** [amzn/ion-js#174]
currently writing symbol logic is below
```
    private writeSymbolToken(s: string) : void {
        if (isIdentifier(s)) {
            if(s.length > 1 && s.charAt(0) === '$'.charAt(0)){
                let tempStr = s.substr(1, s.length);
                if (+tempStr === +tempStr) {//+str === +str is a one line is integer hack
                    this.writeable.writeByte(CharCodes.SINGLE_QUOTE);
                    this.writeable.writeStream(escape(s, SymbolEscapes));
                    this.writeable.writeByte(CharCodes.SINGLE_QUOTE);
                } else {
                    this.writeUtf8(s);
                }
            } else {
                this.writeUtf8(s);
            }
        } else if ((!this.isTopLevel) && (this.currentContainer.containerType === TypeCodes.SEXP) && isOperator(s)) {
            this.writeUtf8(s);
        } else {
            this.writeable.writeByte(CharCodes.SINGLE_QUOTE);
            this.writeable.writeStream(escape(s, SymbolEscapes));
            this.writeable.writeByte(CharCodes.SINGLE_QUOTE);
        }
    }
```
We need to change its name to writeSymbolValue and add another function
```
    private writeSymbolToken(s: string) : void {
                if (s === "'false'" || s === "'true'" || s === "'null'") {
                    this.writeUtf8(s);
                } else {
                    this.writeSymbolValue(s);
                }
    }
```
writeSymbol() should use writeSymbolValue() & writeAnnotations(), writeFieldName() should use writeSymbolToken()

API's can call 
```writeAnnotations(["'false'"]) ```
for printing valid ion. This preserves bidirectional reader-writer functionality.

Here API's are using javascript functionality of equivalent representation of single & double quotes to send them in string without adding escape characters. They can also use literal i.e. backtick
```writeAnnotations([`'false'`])```

**Comment by wesboyt** [comment]
We cant allow the boolean false in all places where symbol false gets called. We only guarantee data model equivalence roundtrip.

symbol tokens and import descriptors have issues tracking them.

**Comment by raganhan** [comment]
As Wes mentioned there is a bug here when using `writeAnnotations` passing either `"false"` or `"true"`. 

`writeAnnotations` contract takes in a array of strings with the textual representation of symbols so:
* `writeAnnotations(["false"])` should represents annotating with the symbol `'false'` 
* `writeAnnotations(["'false'"])` should represents annotating with the symbol `'\'false\''`

`false::1` is not valid Ion but it is the result of `writeAnnotations(["false"])` so there is a bug. Opened https://github.com/amzn/ion-js/issues/206 to track it

**Comment by pbcornell** [comment]
#206 is resolved by #263, and I believe it also resolves this issue.  I'm closing it;  feel free to re-open if I've missed some aspect.  The fix will be part of the next release.  Alternatively, it is merged into the development branch as of now.
